### DISCUSSION TITLE
Is it possible to hide an open tooltip if anchor moves?

### DISCUSSION THREAD

**Discussion by amanharwara** [ariakit/ariakit#2936] [Questions] [answered]
I have a toolbar that can overflow at certain sizes and the items in it have tooltips. If I scroll the toolbar (using Shift + Mouse wheel), the tooltip keeps following the anchor and for some items causes it to go off the screen. Is there a way to hide the tooltip if the anchor moves, or at least stop the tooltip from following the anchor?

**Comment by diegohaz** [comment]
> I scroll the toolbar (using Shift + Mouse wheel)

I'm not sure I understand that behavior. Is that the native browser scroll or a custom event? Either way, you can hide the tooltip when that event happens.

**Reply by amanharwara** [reply]
Hi, this is not a custom event, its just the native browser scroll. Shift + mouse wheel because its a horizontal scroll, not a vertical scroll. So is the only way to fix this to pass the ref of the parent that scrolls to the tooltip and listen to the `scroll` event?

**Reply by diegohaz** [reply, marked as answer]
Yes, but you don't necessarily need to pass the scrolling element ref to the tooltip. You can get it from an element within. Here's the function I use for that purpose:

```ts
/**
 * Returns the scrolling container element of a given element.
 */
export function getScrollingElement(
  element?: Element | null,
): HTMLElement | Element | null {
  if (!element) return null;
  if (element.clientHeight && element.scrollHeight > element.clientHeight) {
    const { overflowY } = getComputedStyle(element);
    const isScrollable = overflowY !== "visible" && overflowY !== "hidden";
    if (isScrollable) return element;
  } else if (element.clientWidth && element.scrollWidth > element.clientWidth) {
    const { overflowX } = getComputedStyle(element);
    const isScrollable = overflowX !== "visible" && overflowX !== "hidden";
    if (isScrollable) return element;
  }
  return (
    getScrollingElement(element.parentElement) ||
    document.scrollingElement ||
    document.body
  );
}
```
